core/meterer/paymentlogic/paymentlogic.go

```diff
@@ -0,0 +1,155 @@
+package paymentlogic
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"slices"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/core"
+)
+
+// GetReservationBinLimit returns the bin limit for a given reservation
+// Note: This is called per-quorum since reservation is for a single quorum.
+func GetReservationBinLimit(reservation *core.ReservedPayment, reservationWindow uint64) uint64 {
+	return GetBinLimit(reservation.SymbolsPerSecond, reservationWindow)
+}
+
+// GetBinLimit returns the bin limit given the bin interval and the symbols per second
+func GetBinLimit(symbolsPerSecond uint64, binInterval uint64) uint64 {
+	return symbolsPerSecond * binInterval
```

Comment:
Any concern about overflow? I think it's unlikely but not sure what we expect large values of symbolsPerSecond and binInterval to be.